src/libraries/System.Private.CoreLib/src/System/IO/Path.Unix.cs

```diff
@@ -17,13 +17,14 @@ public static partial class Path
         private static bool ExistsCore(string fullPath)
         {
             bool result = Interop.Sys.LStat(fullPath, out Interop.Sys.FileStatus fileInfo) == Interop.Errors.ERROR_SUCCESS;
-            if (PathInternal.IsDirectorySeparator(fullPath[fullPath.Length - 1]))
+
+            if (result && PathInternal.IsDirectorySeparator(fullPath[fullPath.Length - 1]))
```

Comment:
could you please also remove the code duplication and move this logic to `Path.Exists`?

`Path.ExistsCore` could return a value tuple `(bool exists, bool isDirectory)` or have an additional `out bool isDirectory` parameter

Reply:
I'm not sure I understand what you mean?
What would the idea of returning if the path is a directory be, since that is not being used inside the Path.Exists function?

Reply:
currently the following logic is duplicated for both Windows and Unix:

```cs
if (result && PathInternal.IsDirectorySeparator(fullPath[fullPath.Length - 1]))
{
     result = isDirectory;
}
```

this check could be moved to `File.Exists` (the method that calls the methods that you have modified in this PR)

Reply:
The only code that is shared is the if statement, and since the part inside the if statement is dependent on stuff defined before the if statement, I don't think it is practical to try and move it into the normal Path.Exists.

Reply:
It's not just about the `if`, it's also about duplicating the comments and keeping both versions in sync.

PTAL at the following commit: https://github.com/dotnet/runtime/commit/fd55a4d8797dff81bed96562a4e35275e3800185

Reply:
I see - you are indeed correct.

I have integrated that into my branch and pushed it again.